pkg/v1/remote/transport/error.go

```diff
@@ -30,13 +30,23 @@ type Error struct {
 	StatusCode int
 	// The raw body if we couldn't understand it.
 	rawBody string
+	// The request that failed
+	request *http.Request
 }
 
 // Check that Error implements error
 var _ error = (*Error)(nil)
 
 // Error implements error
 func (e *Error) Error() string {
+	prefix := ""
+	if e.request != nil {
+		prefix = fmt.Sprintf("%s %s: ", e.request.Method, e.request.URL)
```

Comment:
Will this potentially include URL params related to auth?

Reply:
Possibly -- GCR includes an access token in the query parameters when redirecting to GCS (unless you're using a json key to authenticate, in which case it uses signed URLs). It's a short-lived token so the window is pretty small.

I'm also not sure if this would really be a problem. The http client issues a `GET /v2/<whatever>/blobs/<sha>` and the handles the redirect internally. The godoc for `http.Response.Request` says:

>     // This is only populated for Client requests.

So I'm guessing this is just the original request, not the redirected one.

Reply:
Alright just tested this and it does set `Request` to the redirected URL, do you think we should drop query parameters?

Reply:
Let's maintain a whitelist of expected query string keys from the distribution spec and REDACT anything else to avoid leaking tokens or signed URLs.

Reply:
Done.